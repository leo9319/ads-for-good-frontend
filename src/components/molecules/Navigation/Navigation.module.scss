@import '../../../styles/breakpoints';
@import '../../../styles/colours';

.headerWrapper {
  display: flex;
  align-items: center;
  min-height: 56px;
  padding: 0px var(--spacing-md);
  box-sizing: border-box;
  background-color: $color-white;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.1);

  .logo {
    width: 115px;
    height: 24px;
  }

  .clickable {
    cursor: pointer;
  }
}

@media (max-width: $xs-min) {
  .headerWrapper {
    .logo {
      width: 32px;
      height: 32px;
    }
  }
}
