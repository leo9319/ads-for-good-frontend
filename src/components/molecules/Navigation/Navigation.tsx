import React, { useEffect, useState } from 'react';
import Image from '@components/atoms/Image';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import FalbackImage from '@assets/images/FallbackImage.svg';
import WVLogoDesktop from '@assets/icons/logo/LogoDark.svg';
import WVLogoMobile from '@assets/icons/logo/LogoLightSmall.svg';

import styles from './Navigation.module.scss';

export interface NavigationProps {
  /**
   * Optional className for the header component.
   */
  className?: string;
  /**
   * Optional, header logo alt text.
   * @default 'logo'
   */
  logoAltText?: string;
  /**
   * Optional onClick handler for the logo.
   * @returns void
   */
  onLogoClick?: () => void;
}

export const Navigation = ({
  className,
  logoAltText = 'logo',
  onLogoClick,
}: NavigationProps) => {
  const classes = getModuleClasses(className?.trim(), styles);
  const [isMobile, setIsMobile] = useState(false);
  useEffect(() => {
    const handleResize = () => {
      setIsMobile(window.innerWidth <= 520);
    };

    window.addEventListener('resize', handleResize);
    handleResize();

    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []);
  return (
    <div className={classNames(styles.headerWrapper, classes)}>
      <div>
        <Image
          onClick={onLogoClick}
          src={isMobile ? WVLogoMobile : WVLogoDesktop}
          alt={logoAltText}
          fallbackSrc={FalbackImage}
          className={classNames(
            styles.logo,
            onLogoClick ? styles.clickable : ''
          )}
        />
      </div>
    </div>
  );
};

export default Navigation;
