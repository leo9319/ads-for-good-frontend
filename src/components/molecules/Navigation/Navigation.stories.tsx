import type { Meta, StoryObj } from '@storybook/react';
import Navigation from './Navigation';

const meta = {
  title: 'Library/Molecules/Navigation',
  component: Navigation,
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    className: {
      description: 'Class Name',
      control: {
        type: 'text',
      },
    },
    logoAltText: {
      description: 'Logo Alt Text',
      control: {
        type: 'text',
      },
    },
  },
  args: {
    className: '',
    logoAltText: 'logo',
  },
} satisfies Meta<typeof Navigation>;

export default meta;
type Story = StoryObj<typeof meta>;

export const HeaderWithLogo: Story = {
  args: {},
};
