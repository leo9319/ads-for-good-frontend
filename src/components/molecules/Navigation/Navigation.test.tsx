import React from 'react';
import { render } from '@testing-library/react';
import Navigation from './Navigation';

describe('Navigation', () => {
  it('Should render successfully', () => {
    const { baseElement } = render(<Navigation />);
    expect(baseElement).toMatchSnapshot();
  });
});
