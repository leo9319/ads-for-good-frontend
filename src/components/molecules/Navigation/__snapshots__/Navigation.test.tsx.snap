// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should render successfully 1`] = `
<body>
  <div>
    <div
      class="headerWrapper"
    >
      <div>
        <img
          alt="logo"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper logo"
          inert=""
          src="SvgrURL"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</body>
`;
